### PR TITLE
Addition of xunit project to extend testing capabilities

### PR DESCRIPTION
**Pull request by adaggarwal**
Added unit test project for book.cs
Added a class Statistics.cs to encapsulate the Stats DS
Refatored code for testability

### CODE REVIEW

File: gradebook/src/GradeBook/Statistics.cs [hunk 1]
```diff
@@ -0,0 +1,9 @@
+namespace GradeBook
+{
+    public class Statistics
+    {
+        public double Avg;
```
Comment (crummel) [inline gradebook/src/GradeBook/Statistics.cs:5]: We generally prefer properties to fields.  Changing a property to do additional work (e.g. caching, calculations) is a non-breaking change, but changing a field to do the same is a breaking change.

File: gradebook/src/GradeBook/Statistics.cs [hunk 2]
```diff
@@ -0,0 +1,9 @@
+namespace GradeBook
+{
+    public class Statistics
```
Comment (crummel) [inline gradebook/src/GradeBook/Statistics.cs:3]: I don't think anyone else should be creating instances of `Statistics` - I would add an `internal` constructor to keep them from doing so, or a `private` constructor and a `static` method to create an instance.

File: gradebook/test/GradeBook.tests/BookTest.cs [hunk 3]
```diff
@@ -8,7 +8,19 @@ public class BookTest
...
+            book.AddGrade(40);
 
+            //act
+            var result = book.CalculateStats();
+
+            //assert
+            Assert.Equal(30, result.Avg, 1);
+            Assert.Equal(40, result.High, 1);
+            Assert.Equal(20,result.Low,1);
         }
```
Comment (crummel) [inline gradebook/test/GradeBook.tests/BookTest.cs:24]: - I would add more tests for edge cases. (e.g. negative grade, grade = 0, grade = 100, etc).  You can use [`Assert.Throws`](url) for some of those.
- There's also no testing of `double`s - this would miss incorrect rounding or integer truncation.
- Precision could be less than 1 - generally this should be around the floating point delta for the range you're testing in, so I'd put this at about `0.000001` here.  You also know that `High` and `Low` should be precise (since they're returning just what you passed in), so those should have zero tolerance.
- Inconsistent spacing.

**Review by crummel — CHANGES_REQUESTED** [review]
I like the changes you made for testability.  Just a couple of nitpicks on the new class and a couple other test cases I think you could consider.
